FindWithScalarFields/src/FindWithScalarFields.java

```diff
@@ -0,0 +1,207 @@
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import com.mongodb.BasicDBObject;
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientURI;
+import com.mongodb.client.MongoCollection;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.model.Projections;
+
+import static com.mongodb.client.model.Filters.eq;
+import static com.mongodb.client.model.Filters.and;
+import static com.mongodb.client.model.Filters.gt;
+import static com.mongodb.client.model.Filters.lt;
+import static com.mongodb.client.model.Filters.gte;
+import static com.mongodb.client.model.Filters.lte;
+import static com.mongodb.client.model.Projections.fields;
+import static com.mongodb.client.model.Projections.include;
+import static com.mongodb.client.model.Projections.excludeId;
+
+public class FindWithScalarFields {
+	
+	/**
+	 * Retrieves documents filtered by shipping address postal.
+	 * @param connectionString: To MongoDB instance/MongoDB Cluster.
+	 * @param postalcode: Shipping address postal code to filter documents.
+	 * @return List<Document>: ArrayList of matching documents.
+	 */
+	public List<Document> GetOrdersByPostalCode(String connectionString, int postalcode){
+		if(connectionString == null || connectionString.isEmpty() || postalcode<0)
+		{
+			throw new IllegalArgumentException();
+		}
+		else
+		{
+			MongoClientURI clientUri = new MongoClientURI(connectionString);
```

Comment:
Not sure it makes sense to create a variable for the uri. What made you decide to do that rather than simply embedding the call to MongoClientURI() in the call to the constructor for MongoClient on the next line. This is a coding style comment, fyi.

Reply:
I just wanted to stay inline with the already available documentation on mongoDB website. https://docs.atlas.mongodb.com/driver-connection/#java-driver-example

Shall I change this by excluding MongoClientUri creation?